core/services/pipeline/task.eth_tx.go

```diff
@@ -117,11 +117,35 @@ func (t *ETHTxTask) Run(_ context.Context, lggr logger.Logger, vars Vars, inputs
 		return Result{Error: errors.Wrapf(ErrBadInput, "txMeta: %v", err)}, runInfo
 	}
 
-	err = decoder.Decode(txMetaMap)
+	err = metaDecoder.Decode(txMetaMap)
 	if err != nil {
 		return Result{Error: errors.Wrapf(ErrBadInput, "txMeta: %v", err)}, runInfo
 	}
 
+	var transmitChecker bulletprooftxmanager.TransmitCheckerSpec
+	checkerDecoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
```

Comment:
This is kinda messy, can we move this to a function somewhere? Also would be nice to see unit testing on this

Reply:
Done, moved the tx meta decoding into a separate func too. The first  case of task.eth_tx_test.go tests with a transmit checker, anything else you think would be good to add here?